napari/layers/_pyramid_layer/model.py

```diff
@@ -0,0 +1,249 @@
+
+    @property
+    def pyramid(self):
+        """list: List of np.ndarry image data, with base of pyramid at `0`.
+        """
+        return self._pyramid
+
+    @pyramid.setter
+    def pyramid(self, pyramid):
+        self._pyramid = pyramid
+
+        self.refresh()
+
+    @property
+    def pyramid_level(self):
+        """int: Level of pyramid to show, with base of pyramid at `0`.
+        """
+        return self._pyramid_level
+
+    @pyramid_level.setter
+    def pyramid_level(self, level):
+        if self._pyramid_level == level:
+            return
+        self._pyramid_level = level
+        self.scale = [self._image_downsamples[self.pyramid_level]] * 2
+        self._top_left = self.find_top_left()
+        self._update_image_from_pyramid()
+        self._update_coordinates()
+        self.refresh()
+
+    @property
+    def top_left(self):
+        """int: Location of top left pixel.
+        """
+        return self._top_left
+
+    @top_left.setter
+    def top_left(self, top_left):
+        if np.all(self._top_left == top_left):
+            return
+        self._top_left = top_left
+        self._update_image_from_pyramid()
+        self._update_coordinates()
+        self.refresh()
+
+    def _update_image_from_pyramid(self):
+        """Updates the image data from the pyramid based on requested tile
+        and pyramid level
+        """
+        if np.any(self._image_shapes[self.pyramid_level] >
+                  self._max_tile_shape):
+            slices = [slice(self._top_left[i], self._top_left[i] +
+                            self._max_tile_shape[i], 1) for i in range(2)]
+            self._image = self.pyramid[self.pyramid_level][tuple(slices)]
+            self.translate = self._top_left[::-1]*self.scale[:2]
+        else:
+            self._image = self.pyramid[self.pyramid_level]
+            self.translate = [0, 0]
+
+    def _get_shape(self):
+        """Shape of base of pyramid
+
+        Returns
+        ----------
+        shape : np.ndarry
+            Shape of base of pyramid
+        """
+        if self.multichannel:
+            return self.pyramid[0].shape[:-1]
+        return self.pyramid[0].shape
+
+    def get_value(self):
+        """Returns coordinates, values, and a string for a given mouse position
+        and set of indices.
+
+        Returns
+        ----------
+        coord : sequence of int
+            Position of mouse cursor in data.
+        value : int or float or sequence of int or float
+            Value of the data at the coord.
+        msg : string
+            String containing a message that can be used as
+            a status update.
+        """
+        coord = np.round(self.coordinates).astype(int)
+        if self.multichannel:
+            shape = self._image_view.shape[:-1]
+        else:
+            shape = self._image_view.shape
+        coord[-2:] = np.clip(coord[-2:], 0, np.asarray(shape) - 1)
```

Comment:
`shape` will always be a tuple, so you should use `np.array` or even better, since you're purely using this to achieve broadcasting, I would use either a list comprehension or `np.subtract`

Reply:
fixed with `np.subtract`